plugin.php

```diff
@@ -84,9 +84,12 @@ function run_all_checks( $args ) {
 	}
 
 	if ( ! $args['readme'] ) {
-		$readme_files = preg_grep( '!(^|/)readme.(txt|md)$!i', $top_level_files );
-		if ( $readme_files ) {
-			$args['readme'] = new Readme_Parser( array_shift( $readme_files ) );
+		$readme_txt = preg_grep( '!(^|/)readme.txt$!i', $top_level_files );
+		$readme_md  = preg_grep( '!(^|/)readme.md$!i', $top_level_files );
+		if ( ! empty( $readme_txt ) ) {
```

Comment:
These changes look good to me. I'd remove some of the complexity here with the if/elseif conditions, but it's more a personal preference. I would be fine with merging this in as is.

An example:
```php
$readme_txt     = preg_grep( '!(^|/)readme.txt$!i', $top_level_files );
$readme_md      = preg_grep( '!(^|/)readme.md$!i', $top_level_files );
$args['readme'] = ! empty( $readme_txt ) ? new Readme_Parser( end( $readme_txt ) ) : ( ! empty( $readme_md ) ? new Readme_Parser( end( $readme_md ) ) : null );
```